var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -285,6 +288,21 @@ def link_tau_arch_dirs(self):
                 if os.path.isdir(src) and not os.path.exists(dest):
                     os.symlink(join_path(subdir, d), dest)
 
+    def fix_tau_compilers(self):
+        filter_file('FULL_CC=' + spack_cc, 'FULL_CC=' + self.compiler.cc,
+                    self.prefix + '/include/Makefile', backup=False)
+        filter_file('FULL_CXX=' + re.escape(spack_cxx), 'FULL_CXX=' +
```

Comment:
Why is `re.escape` necessary for this line but not the previous line? You can use `string=True` if you don't want to worry about regexes.

Reply:
The regex was specifically having issues with the ++ in c++ compiler names. I just set string=True for all of these since it should always be literal.